hooks/hiero_update_shot.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+#  UI Hook
+# ===========================
+
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class HieroUpdateShot(HookBaseClass):
+    """
+    This class defines methods that handle updating the Shot entity
+    in Shotgun, as well as whether and how the filesystem structure
+    is created for a Shot during export.
+    """
+    def create_filesystem_structure(self, entity_type, entity_id, preset_properties):
+        """
+        Handles creating the filesystem structure for the shot that
+        was exported. The preset properties dictionary is provided to
+        allow for the lookup of any custom properties that might have
+        been defined in other hooks, like can be achieved when using
+        the hiero_customize_export_ui hook.
+
+        :param str entity_type: The entity type that was created or
```

Comment:
If we're breaking this out into a hook, I'm wondering if we should deemphasize "Shot" to some extent. Maybe something as simple as renaming the hook/class/methods to not include the word "shot".

Reply:
I'm leaning towards not changing this from "shot", simply because it's more or less paired with the existing hiero_get_shot hook. We can't easily rename that hook without potentially breaking client setups that make use of that hook, so I'm inclined to keep get/update shot hooks sharing nomenclature.